src/modules/commander/failsafe/failsafe.cpp

```diff
@@ -0,0 +1,531 @@
+		_rc_lost_at_arming = false;
+	}
+
+	const bool rc_loss_ignored_mission = state.user_intended_mode == vehicle_status_s::NAVIGATION_STATE_AUTO_MISSION
+					     && (_param_com_rcl_except.get() & (int)RCLossExceptionBits::Mission);
+	const bool rc_loss_ignored_loiter = state.user_intended_mode == vehicle_status_s::NAVIGATION_STATE_AUTO_LOITER
+					    && (_param_com_rcl_except.get() & (int)RCLossExceptionBits::Hold);
+	const bool rc_loss_ignored_offboard = state.user_intended_mode == vehicle_status_s::NAVIGATION_STATE_OFFBOARD
+					      && (_param_com_rcl_except.get() & (int)RCLossExceptionBits::Offboard);
+	const bool rc_loss_ignored_takeoff = (state.user_intended_mode == vehicle_status_s::NAVIGATION_STATE_AUTO_TAKEOFF ||
+					      state.user_intended_mode == vehicle_status_s::NAVIGATION_STATE_AUTO_VTOL_TAKEOFF)
+					     && (_param_com_rcl_except.get() & (int)RCLossExceptionBits::Hold);
+	const bool rc_loss_ignored = state.user_intended_mode == vehicle_status_s::NAVIGATION_STATE_AUTO_RTL ||
+				     state.user_intended_mode == vehicle_status_s::NAVIGATION_STATE_AUTO_LAND ||
+				     state.user_intended_mode == vehicle_status_s::NAVIGATION_STATE_AUTO_PRECLAND ||
+				     rc_loss_ignored_mission || rc_loss_ignored_loiter || rc_loss_ignored_offboard || rc_loss_ignored_takeoff ||
+				     ignore_link_failsafe || _rc_lost_at_arming;
+
+	if (_param_com_rc_in_mode.get() != 4 && !rc_loss_ignored) {
+		CHECK_FAILSAFE(status_flags, rc_signal_lost,
+			       fromNavDllOrRclActParam(_param_nav_rcl_act.get()).causedBy(Cause::RCLoss));
+	}
+
+	// Datalink loss
+	const bool data_link_loss_ignored = state.user_intended_mode == vehicle_status_s::NAVIGATION_STATE_AUTO_LAND ||
+					    state.user_intended_mode == vehicle_status_s::NAVIGATION_STATE_AUTO_PRECLAND || ignore_link_failsafe;
+
+	if (_param_nav_dll_act.get() != 0 && !data_link_loss_ignored) {
+		CHECK_FAILSAFE(status_flags, data_link_lost,
+			       fromNavDllOrRclActParam(_param_nav_dll_act.get()).causedBy(Cause::DatalinkLoss));
+	}
+
+	// VTOL transition failure (quadchute)
+	if (state.user_intended_mode == vehicle_status_s::NAVIGATION_STATE_AUTO_MISSION ||
+	    state.user_intended_mode == vehicle_status_s::NAVIGATION_STATE_AUTO_LOITER ||
+	    state.user_intended_mode == vehicle_status_s::NAVIGATION_STATE_AUTO_TAKEOFF ||
+	    state.user_intended_mode == vehicle_status_s::NAVIGATION_STATE_AUTO_VTOL_TAKEOFF) {
+		CHECK_FAILSAFE(status_flags, vtol_transition_failure, fromQuadchuteActParam(_param_com_qc_act.get()));
+	}
+
+	// Mission
+	if (state.user_intended_mode == vehicle_status_s::NAVIGATION_STATE_AUTO_MISSION) {
+		CHECK_FAILSAFE(status_flags, mission_failure, Action::RTL);
+
+		// If RC loss and datalink loss are disabled and we lose both command links and the mission finished,
+		// trigger RTL to avoid losing the vehicle
+		if ((_param_com_rc_in_mode.get() == 4 || rc_loss_ignored_mission) && _param_nav_dll_act.get() == 0
```

Comment:
It would be nice if we could minimize this kind of special logic. How about simply switching into HOLD once mission_finished? 

Ultimately rather than fully opting out of data link loss for mission I think being able to configure relaxed requirements specific to this mode would be safer and satisfy the original intent.

Reply:
I'm all up for simplifying these things as well. Same reason as above though: I did not want to change too much.